File: desktop/backend/desktop_backend/schemas/analytics.py
from __future__ import annotations
from typing import List, Optional
from pydantic import BaseModel


class ModelCapabilities(BaseModel):
    vision: bool = False
    function_calling: bool = False
    streaming: bool = True


class ModelUsageStat(BaseModel):
    provider: str
    model: str
    display_name: Optional[str] = None
    session_count: int
    input_tokens: int
    output_tokens: int
    total_tokens: int
    cost_usd: float
    last_used_at: Optional[str] = None
    capabilities: ModelCapabilities = ModelCapabilities()


class UsageTotals(BaseModel):
    total_sessions: int
    total_input_tokens: int
    total_output_tokens: int
    total_tokens: int
    total_cost_usd: float


class ModelAnalyticsResponse(BaseModel):
    models: List[ModelUsageStat]
    totals: UsageTotals
    period_days: int
    generated_at: str

File: desktop/src/types/analytics.ts
export interface ModelCapabilities {
  vision: boolean;
  function_calling: boolean;
  streaming: boolean;
}

export interface ModelUsageStat {
  provider: string;
  model: string;
  display_name?: string | null;
  session_count: number;
  input_tokens: number;
  output_tokens: number;
  total_tokens: number;
  cost_usd: number;
  last_used_at?: string | null;
  capabilities: ModelCapabilities;
}

export interface UsageTotals {
  total_sessions: number;
  total_input_tokens: number;
  total_output_tokens: number;
  total_tokens: number;
  total_cost_usd: number;
}

export interface ModelAnalyticsResponse {
  models: ModelUsageStat[];
  totals: UsageTotals;
  period_days: number;
  generated_at: string;
}

export type AnalyticsPeriod = 7 | 30 | 90;
